rest-service/manager_rest/rest/filters_utils.py

```diff
@@ -1,129 +1,138 @@
-import re
+from typing import List
 
 from flask import request
 
-from manager_rest import manager_exceptions
+from cloudify._compat import text_type
+
 from manager_rest.rest.rest_utils import validate_inputs
+from manager_rest.manager_exceptions import BadFilterRule, BadParametersError
 from manager_rest.storage import get_storage_manager, models
-from manager_rest.constants import EQUAL, NOT_EQUAL, IS_NULL, IS_NOT_NULL
-
+from manager_rest.constants import (ATTRS_OPERATORS,
+                                    FilterRuleType,
+                                    LabelsOperator,
+                                    LABELS_OPERATORS,
+                                    FILTER_RULE_TYPES)
 
-class BadLabelsFilter(manager_exceptions.BadParametersError):
-    def __init__(self, labels_filter_value):
-        super(BadLabelsFilter, self).__init__(
-            'The labels filter `{0}` is not in the right format. It must be '
-            'one of: <key>=<value>, <key>=[<value1>,<value2>,...], '
-            '<key>!=<value>, <key>!=[<value1>,<value2>,...], <key> is null, '
-            '<key> is not null'.format(labels_filter_value)
-        )
 
-
-def get_filter_rules():
+def get_filter_rules(resource_model: models):
```

Comment:
`models` is not the right type, because that's a module. Either `Union[Blueprint, Deployment]` or just SQLResourceBase or some other base class

Reply:
RIght! Changed to `SQLResourceBase` to be more modular.